src/components/pod/pod.test.js

```diff
@@ -13,22 +59,236 @@ const PodWithSoftDelete = () => (
+      ["tertiary", "rgb(237, 241, 242)", "none"],
+      ["tile", "rgb(255, 255, 255)", "rgba(2, 18, 36, 0.2) 0px 2px 3px 0px"],
+      ["transparent", "rgba(0, 0, 0, 0)", "none"],
+    ])(
+      "should check for %s variant pod, the color is %s and boxShadow is %s",
+      (variant, color, boxShadow) => {
+        CypressMountWithProviders(<PodComponent variant={variant} />);
+        divPodComponent()
+          .should(
+            "have.css",
+            "back-ground",
+            `${color} none repeat scroll 0% 0% / auto padding-box border-box`
+          )
+          .and("have.css", "box-shadow", boxShadow);
+      }
+    );
+
+    it.each(specialCharacters)(
+      "should check title as %s for Pod component",
+      (title) => {
+        CypressMountWithProviders(<PodComponent title={title} />);
+        podTitle().should("have.text", title);
+      }
+    );
+
+    it.each(specialCharacters)(
+      "should check subtitle as %s for Pod component",
+      (subtitle) => {
+        CypressMountWithProviders(<PodComponent subtitle={subtitle} />);
+        podSubTitle().should("have.text", subtitle);
+      }
+    );
+
+    it.each(["left", "center", "right"])(
+      "should check when title alignment is %s for Pod component",
+      (alignTitle) => {
+        CypressMountWithProviders(<PodComponent alignTitle={alignTitle} />);
+        podTitle().should("have.css", "text-align", alignTitle);
+      }
+    );
+
+    it.each(specialCharacters)(
+      "should check footer as %s for Pod component",
+      (footerText) => {
+        CypressMountWithProviders(<PodComponent footer={footerText} />);
+        podFooter().should("have.text", footerText).and("be.visible");
+      }
+    );
+
+    it.each([
+      [true, "be.visible"],
+      [false, "not.exist"],
+    ])(
+      "should check when softDelete is %s for Pod component",
+      (boolVal, state) => {
+        CypressMountWithProviders(
+          <SoftDeletePodComponent softDelete={boolVal} />
+        );
+        podSoftDelete().should(state);
+      }
+    );
+
+    it.each([
+      [false, 77],
+      [true, 1291],
+    ])(
+      "should check when editContentFullWidth is %s for Pod component",
+      (boolVal, width) => {
+        CypressMountWithProviders(
+          <PodComponent editContentFullWidth={boolVal} />
+        );
+        divPodComponent().then(($el) => {
+          useJQueryCssValueAndAssert($el, "width", width);
+        });
+      }
+    );
+
+    it.each([true, false])(
+      "should check when displayEditButtonOnHover is %s for Pod component",
+      (boolVal) => {
+        CypressMountWithProviders(
+          <PodComponent displayEditButtonOnHover={boolVal} />
+        );
+        if (boolVal === true) {
+          podContent().click();
+          podEditContainer().should("be.visible");
+          podDelete().should("not.be.visible");
+        } else {
+          podDelete().should("be.visible");
+        }
+      }
+    );
```

Comment:
```suggestion
    it("edit button is only visible when user clicks on Pod component and displayEditButtonOnHover is true", () => {
        CypressMountWithProviders(
          <PodComponent displayEditButtonOnHover />
        );
        podEditContainer().should("not.be.visible");
        
        podContent().click();
        
        podEditContainer().should("be.visible");
        podDelete().should("not.be.visible");
      }
    );
    
    it("edit and delete buttons are visible when displayEditButtonOnHover is false", () => {
    CypressMountWithProviders(
          <PodComponent displayEditButtonOnHover={false} />
          podEditContainer().should("be.visible");
          podDelete().should("be.visible");
        );
    });
```

In my opinion I would split this test into two to avoid the conditional statement. This would ensure one piece of behaviour is tested per test which would make debugging easier if one of tests begin failing.